pkg/checksum/executor.go

```diff
@@ -252,6 +266,17 @@ func (exec *Executor) Len() int {
 	return len(exec.reqs)
 }
 
+// Each executes the function to each requests in the executor.
+func (exec *Executor) Each(f func(*kv.Request) error) error {
```

Comment:
Could you remove it as it's only used in tests?

Reply:
The test cases are in other packages, hence cannot access private fields that we want to assert...

Keep this function might be necessary, otherwise, we must export the `reqs` field.

Reply:
I see, I prefer to export Reqs directly. `Each` is somewhat unnecessary.

Reply:
I prefer `Each` in some way because it keeps the field being immutable outside. Even definitely it's more complex than exporting this field...